### ISSUE TITLE
'deis pull' generates 500 error 

### ISSUE THREAD

**Issue by arschles** [deis/workflow#69]
the `ultima-windmill` app exists:

```console
core@micro-kube ~/example-go $ ./deis apps
=== Apps
ultima-windmill
```

then, the `deis pull` fails:

```console
core@micro-kube ~/example-go $ ./deis pull deis/example-go -a ultima-windmill
Creating build... Error: 
500 INTERNAL SERVER ERROR
<h1>Server Error (500)</h1>
```

**Comment by arschles** [comment]
here's what looks to be the relevant `deis-workflow` log output:

```console
INFO ultima-windmill: build ultima-windmill-659474b created
INFO ultima-windmill: release ultima-windmill-v2 created
INFO [ultima-windmill]: arschles deployed deis/example-go
INFO Pulling Docker image deis/example-go:latest
INFO Tagging Docker image deis/example-go:latest as deis/example-go:latest
INFO Building Docker image 10.3.0.235:5000/ultima-windmill:v2
INFO Pushing Docker image 10.3.0.235:5000/ultima-windmill:v2
INFO ultima-windmill: 500 Server Error: Internal Server Error ("invalid registry endpoint https://10.3.0.235:5000/v0/: unable to ping registry endpoint https://10.3.0.235:5000/v0/
v2 ping attempt failed with error: Get https://10.3.0.235:5000/v2/: tls: oversized record received with length 20527
 v1 ping attempt failed with error: Get https://10.3.0.235:5000/v1/_ping: tls: oversized record received with length 20527. If this private registry supports only HTTP or HTTPS with an unknown CA certificate, please add `--insecure-registry 10.3.0.235:5000` to the daemon's arguments. In the case of HTTPS, if you have access to the registry's CA certificate, no need for the flag; simply place the CA certificate at /etc/docker/certs.d/10.3.0.235:5000/ca.crt")
INFO [ultima-windmill]: 500 Server Error: Internal Server Error ("invalid registry endpoint https://10.3.0.235:5000/v0/: unable to ping registry endpoint https://10.3.0.235:5000/v0/
v2 ping attempt failed with error: Get https://10.3.0.235:5000/v2/: tls: oversized record received with length 20527
 v1 ping attempt failed with error: Get https://10.3.0.235:5000/v1/_ping: tls: oversized record received with length 20527. If this private registry supports only HTTP or HTTPS with an unknown CA certificate, please add `--insecure-registry 10.3.0.235:5000` to the daemon's arguments. In the case of HTTPS, if you have access to the registry's CA certificate, no need for the flag; simply place the CA certificate at /etc/docker/certs.d/10.3.0.235:5000/ca.crt")
INFO ultima-windmill: arschles scaled containers cmd=1
INFO [ultima-windmill]: arschles scaled containers cmd=1
ERROR Internal Server Error: /v2/apps/ultima-windmill/builds/
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/rest_framework/viewsets.py", line 85, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 407, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python2.7/site-packages/rest_framework/views.py", line 404, in dispatch
    response = handler(request, *args, **kwargs)
  File "/app/api/views.py", line 140, in create
    return super(AppResourceViewSet, self).create(request, **kwargs)
  File "/app/api/views.py", line 117, in create
    return super(BaseDeisViewSet, self).create(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/rest_framework/mixins.py", line 21, in create
    self.perform_create(serializer)
  File "/app/api/viewsets.py", line 21, in perform_create
    self.post_save(obj)
  File "/app/api/views.py", line 254, in post_save
    self.release = build.create(self.request.user)
  File "/app/api/models.py", line 728, in create
    self.app.deploy(user, new_release)
  File "/app/api/models.py", line 457, in deploy
    self._default_scale(user, release)
  File "/app/api/models.py", line 501, in _default_scale
    self.scale(user, structure)
  File "/app/api/models.py", line 292, in scale
    if "scale" in dir(self._scheduler):
  File "/app/api/models.py", line 165, in _scheduler
    mod = importlib.import_module(settings.SCHEDULER_MODULE)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/app/scheduler/k8s.py", line 1, in <module>
    from .scheduler import SchedulerClient  # noqa
ImportError: No module named scheduler
10.2.76.1 "POST /v2/apps/ultima-windmill/builds/ HTTP/1.1" 500 27 "Deis Client v1.13.0-dev"
```

**Comment by mboersma** [comment]
`ImportError: No module named scheduler` is an internal error to the workflow controller. This probably broke with #48. I'll see if I can make a fix.

**Comment by mboersma** [comment]
I'm playing serious whack-a-mole trying to fix this--already several rounds of `KeyErrors` and other basic stuff to fix. I would like to revert to the last SHA we knew was working and then rebuld from there, since unraveling the problems looks to be very difficult. Opinions @arschles @bacongobbler @helgi?

**Comment by arschles** [comment]
@mboersma that sounds like a reasonable step

**Comment by bacongobbler** [comment]
@mboersma yes that sounds quite reasonable. Sounds like it was broken recently, as this was working 2 days ago.

**Comment by mboersma** [comment]
Fixed by #70.
